src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Configuration/ConfigurationProjectConfigurationDimensionProvider.cs

```diff
@@ -0,0 +1,111 @@
+    /// below these 3 providers.
+    /// </remarks>
+    [Export(typeof(IProjectConfigurationDimensionsProvider))]
+    [AppliesTo(ProjectCapabilities.ProjectConfigurationsDeclaredDimensions)]
+    [Order(int.MaxValue)]
+    internal class ConfigurationProjectConfigurationDimensionProvider : BaseProjectConfigurationDimensionProvider
+    {
+        [ImportingConstructor]
+        public ConfigurationProjectConfigurationDimensionProvider(IProjectXmlAccessor projectXmlAccessor)
+            : base(projectXmlAccessor, ConfigurationGeneral.ConfigurationProperty, "Configurations")
+        {
+        }
+
+        /// <summary>
+        /// Modifies the project when there's a configuration change.
+        /// </summary>
+        /// <param name="args">Information about the configuration dimension value change.</param>
+        /// <returns>A task for the async operation.</returns>
+        public override async Task OnDimensionValueChangedAsync(ProjectConfigurationDimensionValueChangedEventArgs args)
+        {
+            if (args.DimensionName == _dimensionName)
+            {
+                if (args.Stage == ChangeEventStage.Before)
+                {
+                    switch (args.Change)
+                    {
+                        case ConfigurationDimensionChange.Add:
+                            await OnConfigurationAdded(args.Project, args.DimensionValue).ConfigureAwait(true);
+                            break;
+                        case ConfigurationDimensionChange.Delete:
+                            await OnConfigurationRemoved(args.Project, args.DimensionValue).ConfigureAwait(true);
+                            break;
+                        case ConfigurationDimensionChange.Rename:
+                            // Need to wait until the core rename changes happen before renaming the property.
+                            break;
+                    }
+                }
+                else if (args.Stage == ChangeEventStage.After)
+                {
+                    // Only change that needs to be handled here is renaming configurations which needs to happen after all
+                    // of the core changes to rename existing conditions have executed.
+                    if (args.Change == ConfigurationDimensionChange.Rename)
+                    {
+                        await OnConfigurationRenamed(args.Project, args.OldDimensionValue, args.DimensionValue).ConfigureAwait(true);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Adds a configuration to the project.
+        /// </summary>
+        /// <param name="unconfiguredProject">Unconfigured project for which the configuration change.</param>
+        /// <param name="configurationName">Name of the new configuration.</param>
+        /// <returns>If the method succeeds, it returns S_OK. If it fails, it returns an error code.</returns>
```

Comment:
Good catch, one of the initial implementations was using IVsCfgProvider and that was the doc for that interface. I'll go over all the comments for Add/Remove/Rename and make sure they are updated properly.